maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/report/Requirements.java

```diff
@@ -0,0 +1,94 @@
+package org.apache.maven.plugin.plugin.report;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Properties;
+
+/**
+ * Plugin requirements.
+ *
+ * @deprecated will be removed in the next major version
+ */
+@Deprecated
+public class Requirements
```

Comment:
Wait a second, not the entire class will go, but only some elements, no?

Reply:
Entire class should be deprecated as parameter `requirements` is deprecated.
When we remove parameter `requirements` class will also be removed.

Reply:
Now I understand, the histories replace it altogether.

Reply:
histories will be used manually by the users to point historical requirements. 
Current requirements should be computed based on current project configuration.